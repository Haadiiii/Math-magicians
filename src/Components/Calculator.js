import React, { Component } from "react";

export default class Calculator extends Component {
  render() {
    return (
      <div className="button">
        <button type="submit">{this.props.name}</button>
      </div>
    );
  }
}
