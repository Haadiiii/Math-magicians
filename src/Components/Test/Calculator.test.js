import React from "react";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Calculator from "../Calculator";

afterEach(cleanup);

it("render Without crashing", () => {
  render(<Calculator />);
});

it("matches snapshot for button", () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
