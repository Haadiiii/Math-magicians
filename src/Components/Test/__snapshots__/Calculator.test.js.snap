// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for button 1`] = `
<div
  className="App"
>
  <div
    className="input-line"
  >
    <div
      className="buttons"
    >
      <div
        className="input"
      >
        0
      </div>
    </div>
  </div>
  <div
    className="button-line1 buttons"
  >
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        AC
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        7
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        4
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        1
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        0
      </button>
    </div>
  </div>
  <div
    className="button-line2 buttons"
  >
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        +/-
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        8
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        5
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        2
      </button>
    </div>
  </div>
  <div
    className="button-line3 buttons"
  >
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        %
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        9
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        6
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        3
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        .
      </button>
    </div>
  </div>
  <div
    className="button-line4 buttons"
  >
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        รท
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        x
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        -
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        +
      </button>
    </div>
    <div
      className="button"
    >
      <button
        onClick={[Function]}
        type="submit"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
